### PR TITLE
CLOUDP-175080: Support deletion protection for Atlas projects

### PR DESCRIPTION
**Pull request by helderjs**
### All Submissions:

* [x] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if there is one).
* [ ] Update docs/release-notes/release-notes.md if your changes should be included in the release notes for the next release.

### CODE REVIEW

File: test/int/project_protect_test.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+					WithIPAccessList(project.NewIPAccessList().WithCIDR("0.0.0.0/0"))
+				Expect(k8sClient.Create(context.TODO(), testProject, &client.CreateOptions{})).To(Succeed())
+
+				Eventually(func() bool {
+					return testutil.CheckCondition(k8sClient, testProject, status.TrueCondition(status.ReadyType))
+				}).WithTimeout(15 * time.Minute).WithPolling(PollingInterval).Should(BeTrue())
+			})
+
+			// nolint:dupl
+			By("Deleting project in cluster don't delete from Atlas", func() {
```
Comment (josvazg) [inline test/int/project_protect_test.go:56]:
```suggestion
			By("Deleting project in cluster doesn't delete it from Atlas", func() {
```

File: test/int/project_protect_test.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+					WithConnectionSecret(connectionSecret.Name)
+				Expect(k8sClient.Create(context.TODO(), testProject, &client.CreateOptions{})).To(Succeed())
+
+				Eventually(func() bool {
+					return testutil.CheckCondition(k8sClient, testProject, status.TrueCondition(status.ReadyType))
+				}).WithTimeout(15 * time.Minute).WithPolling(PollingInterval).Should(BeTrue())
+			})
+
+			// nolint:dupl
+			By("Deleting project in cluster don't delete from Atlas", func() {
```
Comment (josvazg) [inline test/int/project_protect_test.go:98]:
```suggestion
			By("Deleting project in cluster doesn't delete it from Atlas", func() {
```

**Comment by josvazg** [comment]
LGTM overall
I wonder whether or not we want more tests covering the annotation cases `keep` vs `delete`.

**Review by josvazg — APPROVED** [review]

File: test/int/project_protect_test.go [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+				Eventually(func(g Gomega) {
+					g.Expect(k8sClient.Get(context.TODO(), client.ObjectKeyFromObject(testProject), testProject, &client.GetOptions{})).ToNot(Succeed())
+
+					atlasProject, _, err := atlasClient.Projects.GetOneProjectByName(context.TODO(), projectName)
+					g.Expect(err).To(BeNil())
+					g.Expect(atlasProject).ToNot(BeNil())
+				}).WithTimeout(5 * time.Minute).WithPolling(PollingInterval).Should(Succeed())
+
+				_, err := atlasClient.Projects.Delete(context.TODO(), projectID)
+				Expect(err).To(BeNil())
```
Comment (igor-karpukhin) [inline test/int/project_protect_test.go:114]: non-blocking: this can be moved to `AfterEach`

**Review by igor-karpukhin — APPROVED** [review]
Good job 👍 Looks good in general!

**Pull request merged**